pennylane/grouping/pauli.py

```diff
@@ -0,0 +1,254 @@
+
+
+def pauli_group(n_qubits, wire_map=None):
+    """Generate the :math:`n`-qubit Pauli group.
+
+    This function enables the construction of the :math:`n`-qubit Pauli group with no
+    storage involved.  The :math:`n`-qubit Pauli group has size :math:`4^n`,
+    thus it may not be desirable to construct it in full and store.
+
+    The order of iteration is based on the binary symplectic representation of
+    the Pauli group as :math:`2n`-bit strings. Ordering is done by converting
+    the integers :math:`0` to :math:`2^{2n}` to binary strings, and converting those
+    strings to Pauli operators using the ``binary_to_pauli`` method.
+
+    Args:
+        n_qubits (int): The number of qubits for which to create the group.
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels
+            used in the Pauli word as keys, and unique integer labels as their values.
+            If no wire map is provided, wires will be labeled by integers between 0 and ``n_qubits``.
+
+    Returns:
+        .Operation: The next Pauli word in the group.
+
+    **Example**
+
+    The ``pauli_group`` generator can be used to loop over the Pauli group as follows:
+
+    .. code-block:: python
+
+        from pennylane.pauli import pauli_group
+
+        n_qubits = 3
+
+        for p in pauli_group(n_qubits):
+            print(p)
+
+    The Pauli group in full can be obtained in full like so:
+
+    .. code-block:: python
+
+       full_pg = list(pauli_group(n_qubits))
+
+    The group can also be created using a custom wire map (if no map is
+    specified, a default map of label :math:`i` to wire ``i`` will be created).
+
+    .. code-block:: python
+
+        n_qubits = 3
+        wire_map = {'a' : 0, 'b' : 1, 'c' : 2}
+
+        for p in pauli_group(n_qubits, wire_map=wire_map):
+            print(p)
+
+    """
+    if not isinstance(n_qubits, int):
+        raise TypeError("Must specify an integer number of qubits construct the Pauli group.")
+
+    if n_qubits <= 0:
+        raise ValueError("Number of qubits must be at least 1 to construct Pauli group.")
+
+    return _pauli_group_generator(n_qubits, wire_map=wire_map)
+
+
+def pauli_mult(pauli_1, pauli_2, wire_map=None):
+    """Multiply two Pauli words together.
+
+    Two Pauli operations can be multiplied together by taking the additive
+    OR of their binary symplectic representations.
+
+    Args:
+        pauli_1 (.Operation): A Pauli word.
+        pauli_2 (.Operation): A Pauli word to multiply with the first one.
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels used in the Pauli
+            word as keys, and unique integer labels as their values. If no wire map is
+            provided, the map will be constructed from the set of wires acted on
+            by the input Pauli words.
+
+    Returns:
+        .Operation: The product of pauli_1 and pauli_2 as a Pauli word
+            (ignoring the global phase).
```

Comment:
```suggestion
        .Operation: The product of pauli_1 and pauli_2 as a Pauli word
        (ignoring the global phase).
```
(return descriptions don't have an indentation on continuation lines; adding one results in a rendering error)